scip_indexer/SCIPIndexer.cc

```diff
@@ -561,25 +585,42 @@ class AliasMap final {
-                                {bind.bind.variable,
-                                 {GenericSymbolRef::undeclaredField(klass, instr->name, bind.bind.type), bind.loc,
-                                  false}});
+                    auto [result, cacheHit] = fieldResolver.findUnresolvedFieldTransitive(
+                        ctx, ctx.locAt(bind.loc), {klass.asClassOrModuleRef(), instr->name});
+                    auto checkExists = [&](bool exists, const std::string &text) {
+                        ENFORCE(exists,
+                                "Returned non-existent {} from findUnresolvedFieldTransitive with start={}, "
+                                "field={}, file={}, loc={}",
+                                text, klass.exists() ? klass.toStringFullName(gs) : "<non-existent>",
+                                instr->name.exists() ? instr->name.toString(gs) : "<non-existent>",
+                                ctx.file.data(gs).path(), ctx.locAt(bind.loc).showRawLineColumn(gs));
+                    };
+                    switch (result.inherited.kind()) {
+                        case FieldQueryResult::Kind::FromUndeclared: {
+                            checkExists(result.inherited.originalClass().exists(), "class");
+                            auto namedSymRef = GenericSymbolRef::undeclaredField(klass, instr->name, bind.bind.type);
+                            if (!cacheHit) {
+                                // It may be the case that the mixin values are already stored because of the
+                                // traversal in some other function. In that case, don't bother overriding.
+                                relMap.insert({namedSymRef.withoutType(), result});
+                            }
+                            // no trim(...) because undeclared fields shouldn't have ::
+                            this->map.insert({bind.bind.variable, {namedSymRef, bind.loc, false}});
+                            break;
                         }
-                    } else if (absl::holds_alternative<core::SymbolRef>(result)) {
-                        auto fieldSym = absl::get<core::SymbolRef>(result);
-                        if (fieldSym.exists()) {
-                            this->map.insert(
-                                {bind.bind.variable,
-                                 {GenericSymbolRef::declaredField(fieldSym, bind.bind.type), trim(bind.loc), false}});
+                        case FieldQueryResult::Kind::FromDeclared: {
+                            auto fieldSym = result.inherited.originalSymbol();
+                            checkExists(fieldSym.exists(), "field");
+                            auto namedSymRef =
+                                fieldSym.owner(gs) == klass
+                                    ? GenericSymbolRef::declaredField(fieldSym, bind.bind.type)
+                                    : GenericSymbolRef::undeclaredField(klass, instr->name, bind.bind.type);
```

Comment:
Little sus on this inconsistency -- now I'm thinking that the only thing that should be changing is what we link it to, not the symbol that we use. I need to figure out how to test the declared case...

Reply:
Yeah, this is definitely wrong. We should consistently use the "current" class and emit a relationship.